### PR TITLE
8303136: MemoryPoolMBean/isCollectionUsageThresholdExceeded/isexceeded005 failed with "isCollectionUsageThresholdExceeded() returned true, while threshold = 1 and used = 0"

### PR DESCRIPTION
**Pull request by kevinjwalls**
Test update for an occasional failure, which does not reproduce.

The test failure in JDK-8303136 is at line 141 in the updated file here.  It's the failure where isExceeded is true, but our sampled "used" value is not above the threshold.  But while the comment says it's refreshing values, it does not not refresh "used", so there could have been gc/promotion activity which hits the threshold outside of the test's control.  Refreshing "used" is the addition here.

Separately, the code at line 123 in the new file also claims to refresh the values, but it only refreshes the threshold, which we aren't changing.  Not making it refresh "used" at that point looks correct, so remove the "if (used >= threshold)" as we have already checked that at line 116.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303136](https://bugs.openjdk.org/browse/JDK-8303136): MemoryPoolMBean/isCollectionUsageThresholdExceeded/isexceeded005 failed with "isCollectionUsageThresholdExceeded() returned true, while threshold = 1 and used = 0"

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12823/head:pull/12823` \
`$ git checkout pull/12823`

Update a local copy of the PR: \
`$ git checkout pull/12823` \
`$ git pull https://git.openjdk.org/jdk pull/12823/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12823`

View PR using the GUI difftool: \
`$ git pr show -t 12823`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12823.diff">https://git.openjdk.org/jdk/pull/12823.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/monitoring/MemoryPoolMBean/isCollectionUsageThresholdExceeded/isexceeded001.java [hunk 1]
```diff
@@ -120,15 +120,11 @@ public static int run(String[] argv, PrintStream out) {
                 isExceeded = monitor.isCollectionThresholdExceeded(pool);
                 if (!isExceeded) {
 
-                    // Refresh the values
-                    threshold = monitor.getCollectionThreshold(pool);
-                    usage = monitor.getCollectionUsage(pool);
-                    if (used >= threshold) {
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/monitoring/MemoryPoolMBean/isCollectionUsageThresholdExceeded/isexceeded001.java:126]: Although `used` is not updated, `threshold` is. Couldn't removing this extra check result in new failures?

Comment (kevinjwalls) [reply test/hotspot/jtreg/vmTestbase/nsk/monitoring/MemoryPoolMBean/isCollectionUsageThresholdExceeded/isexceeded001.java:126]: I think that's OK.  We set the threshold earlier in the test, to the value of used+1, we don't need to read threshold back again.  I see in testing that it reads back the value that it set, as it should, this value isn't changed again.  

(The test's monitor.setCollectionThreshold method (MemoryMonitor.java) for setting this ends up calling into MemoryPoolImpl.java)

The testing at lines 116 - 121 is checking if we have crossed the threshold, and if so then isExceeded really should be true.  It's surviving testing so far!

**Review by plummercj — APPROVED** [review]

**Review by sspitsyn — APPROVED** [review]
Looks good to me.
Thanks,
Serguei

**Comment by kevinjwalls** [comment]
Thanks Chris, thanks Serguei!

**Pull request closed without merging**